src/Redux/Books/Books.js

```diff
@@ -13,12 +15,25 @@ export const removeBook = (payload) => ({
   payload,
 });
 
+export const setBooks = (payload) => ({
+  type: SET_BOOKS,
+  payload,
+});
+
+export const clearBooks = () => ({
+  type: CLEAR_BOOKS,
+});
+
 const reducer = (state = initialState, action) => {
   switch (action.type) {
     case ADD_BOOK:
       return [...state, action.payload];
     case REMOVE_BOOK:
-      return [...state.filter((book) => book.id !== action.payload)];
+      return state.filter((book) => book.item_id !== action.payload);
```

Comment:
- [ ] As you can see from the image below, your book is not being removed from the store. This is because you are using `item_id` here instead of `id`. Kindly fix that.

  This code snippet should work: `return state.filter((book) => book.id !== action.payload);`

  ![ayoub-remove](https://user-images.githubusercontent.com/36542986/160431615-5be1807e-48a0-40f8-9235-6f9ab01da34e.jpg)

- [ ] **[OPTIONAL]** Whenever I add a book to your app I get a warning, I would recommend you fix that (see image below)

  ![ayoub-error](https://user-images.githubusercontent.com/36542986/160432232-246d069e-7904-4d74-b1cc-e97d27a08ed5.jpg)

Reply:
Done